<launch>
    <!-- Global parameters -->
    <param name="starting" type="int" value="151"/> <!-- 61 -->
    <param name="rate" type="double" value="0.1"/>

    <!-- Nodes -->

    <!-- get car info -->
    <node pkg="perceptive_stream" type="gndTruthCAR.py" name="gnd_truth_car0" output="screen">
        <param name="json_path" value="/home/caillot/Documents/Dataset/CARLA_Dataset_A/Embed/V0/VehicleInfo/%06d.json"/>
        <param name="vehicle_ID" value="V0_info."/>
    </node>

    <node pkg="perceptive_stream" type="gndTruthCAR.py" name="gnd_truth_car1" output="screen">
        <param name="json_path" value="/home/caillot/Documents/Dataset/CARLA_Dataset_A/Embed/V1/VehicleInfo/%06d.json"/>
        <param name="vehicle_ID" value="V1_info."/>
    </node>

    <node pkg="perceptive_stream" type="gndTruthCAR.py" name="gnd_truth_car2" output="screen">
        <param name="json_path" value="/home/caillot/Documents/Dataset/CARLA_Dataset_A/Embed/V2/VehicleInfo/%06d.json"/>
        <param name="vehicle_ID" value="V2_info."/>
    </node>

    <!-- get participating sensors -->

    <node pkg="perceptive_stream" type="readCamStream.py" name="dataset_infra_camRGB" output="screen">
        <param name="json_path" value="/home/caillot/Documents/Dataset/CARLA_Dataset_A/Infra/sensorInfo/%06d.json" />
        <param name="npy_path" value="/home/caillot/Documents/Dataset/CARLA_Dataset_A/Infra/cameraRGB/cameraMatrix.npy" />
        <param name="img_path" value="/home/caillot/Documents/Dataset/CARLA_Dataset_A/Infra/cameraRGB/%06d.png" />
        <param name="sensor_ID" value="Infra_camRGB."/>
    </node>

    <node pkg="perceptive_stream" type="readCamStream.py" name="dataset_V0_camRGB" output="screen">
        <param name="json_path" value="/home/caillot/Documents/Dataset/CARLA_Dataset_A/Embed/V0/VehicleInfo/%06d.json" />
        <param name="npy_path" value="/home/caillot/Documents/Dataset/CARLA_Dataset_A/Embed/V0/cameraRGB/cameraMatrix.npy" />
        <param name="img_path" value="/home/caillot/Documents/Dataset/CARLA_Dataset_A/Embed/V0/cameraRGB/%06d.png" />
        <param name="sensor_ID" value="V0_camRGB."/>
    </node>

    <node pkg="perceptive_stream" type="readCamStream.py" name="dataset_V1_camRGB" output="screen">
        <param name="json_path" value="/home/caillot/Documents/Dataset/CARLA_Dataset_A/Embed/V1/VehicleInfo/%06d.json" />
        <param name="npy_path" value="/home/caillot/Documents/Dataset/CARLA_Dataset_A/Embed/V1/cameraRGB/cameraMatrix.npy" />
        <param name="img_path" value="/home/caillot/Documents/Dataset/CARLA_Dataset_A/Embed/V1/cameraRGB/%06d.png" />
        <param name="sensor_ID" value="V1_camRGB."/>
    </node>

    <node pkg="perceptive_stream" type="readCamStream.py" name="dataset_V2_camRGB" output="screen">
        <param name="json_path" value="/home/caillot/Documents/Dataset/CARLA_Dataset_A/Embed/V2/VehicleInfo/%06d.json" />
        <param name="npy_path" value="/home/caillot/Documents/Dataset/CARLA_Dataset_A/Embed/V2/cameraRGB/cameraMatrix.npy" />
        <param name="img_path" value="/home/caillot/Documents/Dataset/CARLA_Dataset_A/Embed/V2/cameraRGB/%06d.png" />
        <param name="sensor_ID" value="V2_camRGB."/>
    </node>

    <!-- Extract 2D bounding box -->

    <node pkg="perceptive_stream" type="BBox2D_publisher.py" name="BBox2D_publisher_inf_1" output="screen">
        <param name="queue_size" type="int" value="100"/>
    </node>

    <!-- Create Local occupancy grid -->

    <node pkg="perceptive_stream" type="bbox2d_to_gol.py" name="bbox2d_to_gol_inf_1" output="screen">
        <param name="queue_size" type="int" value="100"/>
        <param name="o3d" type="bool" value="true"/>
    </node>

    <!-- Aggregate GOL -->

    <node pkg="perceptive_stream" type="GOL_merger.py" name="gol_merger" output="screen">
    </node>

</launch>